packages/material-ui/src/Slider/SliderBase.js

```diff
@@ -0,0 +1,922 @@
+  const handleTouchStart = useEventCallback((event) => {
+    // If touch-action: none; is not supported we need to prevent the scroll manually.
+    if (!doesSupportTouchActionNone()) {
+      event.preventDefault();
+    }
+
+    const touch = event.changedTouches[0];
+    if (touch != null) {
+      // A number that uniquely identifies the current finger in the touch session.
+      touchId.current = touch.identifier;
+    }
+    const finger = trackFinger(event, touchId);
+    const { newValue, activeIndex } = getFingerNewValue({ finger, values, source: valueDerived });
+    focusThumb({ sliderRef, activeIndex, setActive });
+
+    setValueState(newValue);
+
+    if (onChange) {
+      onChange(event, newValue);
+    }
+
+    const doc = ownerDocument(sliderRef.current);
+    doc.addEventListener('touchmove', handleTouchMove);
+    doc.addEventListener('touchend', handleTouchEnd);
+  });
+
+  React.useEffect(() => {
+    const { current: slider } = sliderRef;
+    slider.addEventListener('touchstart', handleTouchStart, {
+      passive: doesSupportTouchActionNone(),
+    });
+
+    const doc = ownerDocument(slider);
+
+    return () => {
+      slider.removeEventListener('touchstart', handleTouchStart, {
+        passive: doesSupportTouchActionNone(),
+      });
+
+      doc.removeEventListener('mousemove', handleTouchMove);
+      doc.removeEventListener('mouseup', handleTouchEnd);
+      doc.removeEventListener('touchmove', handleTouchMove);
+      doc.removeEventListener('touchend', handleTouchEnd);
+    };
+  }, [handleTouchEnd, handleTouchMove, handleTouchStart]);
+
+  React.useEffect(() => {
+    if (disabled) {
+      const doc = ownerDocument(sliderRef.current);
+      doc.removeEventListener('mousemove', handleTouchMove);
+      doc.removeEventListener('mouseup', handleTouchEnd);
+      doc.removeEventListener('touchmove', handleTouchMove);
+      doc.removeEventListener('touchend', handleTouchEnd);
+    }
+  }, [disabled, handleTouchEnd, handleTouchMove]);
+
+  const handleMouseDown = useEventCallback((event) => {
+    if (onMouseDown) {
+      onMouseDown(event);
+    }
+
+    event.preventDefault();
+    const finger = trackFinger(event, touchId);
+    const { newValue, activeIndex } = getFingerNewValue({ finger, values, source: valueDerived });
+    focusThumb({ sliderRef, activeIndex, setActive });
+
+    setValueState(newValue);
+
+    if (onChange) {
+      onChange(event, newValue);
+    }
+
+    const doc = ownerDocument(sliderRef.current);
+    doc.addEventListener('mousemove', handleTouchMove);
+    doc.addEventListener('mouseup', handleTouchEnd);
+  });
+
+  const trackOffset = valueToPercent(range ? values[0] : min, min, max);
+  const trackLeap = valueToPercent(values[values.length - 1], min, max) - trackOffset;
+  const trackStyle = {
+    ...axisProps[axis].offset(trackOffset),
+    ...axisProps[axis].leap(trackLeap),
+  };
+
+  const Root = components.root || 'span';
+  const rootProps = componentsProps.root || {};
```

Comment:
Spreading `undefined` can work too

```suggestion
  const rootProps = componentsProps.root;
```

Reply:
I am also using the `rootProps` and the other for accessing the `className` prop at this moment. Let me see if I can avoid it, if not I would rather leave it here, rather than doing it when accessing a prop.

Reply:
Oh, I didn't realize it, sounds fine then